9491 intermediate to test

ujahio opened the pull request:
Resolves some merge conflicts to test environment

ujahio committed:
9491 - renamed some files and added the correct function names

ujahio committed:
9491 - added the sorting button to my inbox for the docket number column [skip ci]

shumway-tm committed:
9491 - added sortable column headers and sorting capability to other fields in Inbox, Outbox, and Completed

shumway-tm committed:
9491 - genericized intergration test to handle both section and my queues for message boxes

ujahio committed:
9491 - removed the hasMessages compute from the messages jsxs'

ujahio committed:
9491 - added test to verify that sent messages are sorted from newest to oldest for non-adc users

shumway-tm committed:
9491 - dynamically remove tooltip title from inactive sort field icons for voiceover purposes

ujahio committed:
Merge branch 'test' into 9491-intermediate-to-test

ujahio committed:
9491 - updated set default table sort action to account for default sort for invalid box selection

ujahio committed:
Merge branch '9491-sort-my-messages' into 9491-intermediate-to-test

codyseibert committed:
Merge branch '9490-sorting-adc-section-messages-round-two' into 9491-sort-my-messages

shumway-tm committed:
Merge branch '9491-sort-my-messages' into 9491-intermediate-to-test

shumway-tm committed:
Merge branch 'test' into 9491-intermediate-to-test

shumway-tm committed:
9491 - lint

shumway-tm committed:
9491 - resolving merge conflict on cherrypick of completedMessage 7fd308d

shumway-tm committed:
9491 - fixed import in integration test

shumway-tm committed:
9491 - fixed import in integration test

ujahio committed:
9491 - refactored getUserMessageCount helper function in journey story

ujahio committed:
Merge branch '9491-sort-my-messages' into 9491-intermediate-to-test

ujahio committed:
Merge branch '9491-intermediate-to-test' of https://github.com/flexion/ef-cms into 9491-intermediate-to-test

shumway-tm committed:
Merge branch '9491-sort-my-messages' of https://github.com/flexion/ef-cms into 9491-sort-my-messages

rachelschneiderman commented on web-client/integration-tests/adcClerkViewsMessagesJourney.test.js:
```diff
@@ -0,0 +1,296 @@
+      toUserId: testAdcId,
+    });
+
+    loginAs(cerebralTest, 'docketclerk@example.com');
```
Do we need to log in again as 'docketclerk' when we are already logged in on 74?

rachelschneiderman commented on web-client/integration-tests/adcClerkViewsMessagesJourney.test.js:
```diff
@@ -0,0 +1,296 @@
+
+    loginAs(cerebralTest, 'adc@example.com');
+    it('get before counts for all section message boxes', async () => {
+      await getUserMessageCount(cerebralTest, 'inbox', messageQueue);
```
Could we make use of the value returned from getUserMessageCount like on line 109 to get the length of the  messages in the provided messageQueue?

ujahio replied:
Yes. We've made that change.

rachelschneiderman commented on web-client/integration-tests/adcClerkViewsMessagesJourney.test.js:
```diff
@@ -0,0 +1,296 @@
+  });
+});
+
+const validateMessageOrdering = (actualMessages, expectedMessageSubjects) => {
```
💭 Could we make use of the `expect.arrayContaining` method in jest to enumerate what we expect the message ordering to be? That way if the test fails because the messages are not ordered as we expect, it may be more obvious why the sorting is not as we expected.

ujahio replied:
After some research, we think expect.arrayContaining guarantees order. Part of the tests we performed was altering the ideal order and the tests returned a false positive.

rachelschneiderman commented on web-client/integration-tests/adcClerkViewsMessagesJourney.test.js:
```diff
@@ -0,0 +1,296 @@
+  messageCompletedSubject,
+  completedMessageSubject,
+) => {
+  const foundMessage = messages.find(
```
❓ Is there a way to save the `messageId`s from each of the messages created during test setup? That way we can be sure we are grabbing the message we intend off of state. I see that the date is included in the subject, which is a good way of uniquely identifying the messages! Just wondering if there is a more direct way.

ujahio replied:
Yes. There is a way to use ids. We have made the appropriate refactor to use ids.

rachelschneiderman commented on web-client/src/presenter/actions/setDefaultTableSortAction.test.js:
```diff
@@ -65,41 +66,40 @@ describe('setDefaultTableSortAction', () => {
   });
 
-  it('should not set the tableSort if role is not ADC', async () => {
+  it('the completed should be sorted by createdAt DESCENDING for adc user', async () => {
```
I think there's a discrepency between the test name and the expectations here, line 73 says the completed messages should be sorted by `createdAt` but the expectation says `completedAt`.

ujahio replied:
Good catch

rosiehsmith commented on web-client/src/presenter/actions/setDefaultTableSortAction.test.js:
```diff
@@ -65,41 +66,40 @@ describe('setDefaultTableSortAction', () => {
       state: {
         user: {
-          role: 'adc',
+          role: 'docketclerk',
```
Could you explicitly set the tableSort here in the `arrange` portion of the test so we can be sure this will fail? Thanks!

ujahio replied:
Good catch, added the default sort order

rachelschneiderman commented on web-client/src/presenter/actions/setDefaultTableSortAction.js:
```diff
@@ -10,14 +10,13 @@ import { state } from 'cerebral';
   }
 
-  // different tables require different sorting
+  // different tables require different default sorting
```
Nit: y'all added some really awesome tests for this functionality, is this comment needed?

ujahio replied:
Made the change.

rachelschneiderman commented on web-client/src/views/Messages/MessagesIndividualCompleted.jsx:
```diff
@@ -1,21 +1,80 @@
+  ALPHABETICALLY_DESCENDING,
+  CHRONOLOGICALLY_ASCENDING,
+  CHRONOLOGICALLY_DESCENDING,
+} from './sortConstants';
```
❓ Why were these added to a `sortConstants` over adding them to the existing applicationContext constants?

ujahio replied:
No particular reason. We have migrated those items to application context.

codyseibert committed:
9491 - addressing pr comment related to message subjects and to instead us message ids

ujahio committed:
9491 - refactor and changes based on feedback

ujahio committed:
Merge branch '9491-sort-my-messages' into 9491-copy-1

# Conflicts:
#	web-client/integration-tests/adcClerkViewsMessagesJourney.test.js

ujahio committed:
9491 - correcting imports and using constants

ujahio committed:
9491 - correcting more constants import

ujahio committed:
9491 - removed unnccessary constant file

ujahio committed:
Merge branch 'staging' of https://github.com/ustaxcourt/ef-cms into 9491-copy

shumway-tm committed:
9491 - fixed failing integration tests, spy missing

shumway-tm committed:
Merge branch 'test' of https://github.com/ustaxcourt/ef-cms into 9491-copy-merge-conflicts

shumway-tm committed:
9491 - resolve merge conflicts

ujahio committed:
9491 - fixing spyon to check for correct failure points

shumway-tm committed:
9491 - readding spyon

shumway-tm committed:
Merge branch '9491-sort-my-messages' into 9491-intermediate-to-test

ujahio committed:
9491 - fixing integration tests for mocking createMessageInteractor

ujahio committed:
9491 -resolving merge conflict after cherry-picking from an intermediate branch

# Conflicts:
#	web-client/integration-tests/adcClerkViewsMessagesJourney.test.js
#	web-client/integration-tests/messagesJourney.test.js

ujahio committed:
9491 - added default sort order to address feedback about forcing a failure when it occurs

ujahio committed:
Merged commit to address adding a default sort order
